gensim/models/_fasttext_bin.py

```diff
@@ -197,6 +194,41 @@ def _load_vocab(fin, new_format, encoding='utf-8'):
     return raw_vocab, vocab_size, nwords
 
 
+def _decode_bad_unicode_py2(word_bytes, encoding):
```

Comment:
A docstring would be welcome; magic happening here.

Reply:
Is it really necessary? My opinion is no, because:

1. It’s an internal function. The docstring will not appear in the generated documentation. According to PEP8, docstrings are not necessary in this case (https://stackoverflow.com/questions/7687407/are-docstrings-for-internal-functions-python-necessary#7687430).
2. The function body is shorter than what a docstring would be.
3. The docstring would not add any new value: it would merely duplicate the function name and body and existing comment.
4. The function behavior is fully captured by unit tests.

Reply:
I'm still +1 on a docstring. Not for the code (which is clean and simple enough, no problem reading that), but for the purpose and motivation behind having this magic in the first place.

If I saw code like this in a new code base, my first reaction would be "WTF, why shouldn't I remove this?"

How about this:

```python
def _decode_bad_unicode_py2(word_bytes, encoding):
    """
    Decode bytes that failed to cleanly decode using utf8, using latin1 instead.

    This is a hack, but it's needed because we're keep *some* representation even of invalid
    (badly encoded) terms. Such invalid terms appear in models produced by other tools, such
    as Facebook's own fastText models (ha!).

    Note the latin1-decoded output is unlikely to be of much use: the characters will be garbage, and
    never actually match with real words. Still, some (ASCII) ngram substrings may be useful, plus
    we avoid term collisions that would otherwise happen with `errors="replace"`, or vocabulary count
    mismatches that would happen if we ignored this invalid term altogether (see also
    https://github.com/RaRe-Technologies/gensim/issues/2402).

    """
    word = word_bytes.decode('latin1')
    logger.error(
        'failed to decode invalid utf8 bytes %r; falling back to latin1, using %r instead',
        word_bytes, word,
    )
    return word
```

Reply:
I ended up reverting this in favor of the backport.